plugins/modules/mso_schema_template_external_epg_selector.py

```diff
@@ -0,0 +1,244 @@
+  delegate_to: localhost
+
+- name: Query a specific Selector
+  cisco.mso.mso_schema_template_anp_epg_selector:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema 1
+    template: Template 1
+    external_epg: extEPG 1
+    selector: selector_1
+    state: query
+  delegate_to: localhost
+  register: query_result
+
+- name: Query all Selectors
+  cisco.mso.mso_schema_template_anp_epg_selector:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema 1
+    template: Template 1
+    external_epg: extEPG 1
+    state: query
+  delegate_to: localhost
+  register: query_result
+'''
+
+RETURN = r'''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec, mso_contractref_spec, issubset, mso_expression_spec_ext_epg
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        schema=dict(type='str', required=True),
+        template=dict(type='str', required=True),
+        external_epg=dict(type='str', required=True),
+        selector=dict(type='str'),
+        expressions=dict(type='list', options=mso_expression_spec_ext_epg()),
+        state=dict(type='str', default='present', choices=['absent', 'present', 'query']),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ['state', 'absent', ['selector']],
+            ['state', 'present', ['selector']],
+        ],
+    )
+
+    schema = module.params.get('schema')
+    template = module.params.get('template')
+    external_epg = module.params.get('external_epg')
+    selector = module.params.get('selector')
+    expressions = module.params.get('expressions')
+    state = module.params.get('state')
+
+    mso = MSOModule(module)
+
+    # Get schema_id
+    schema_obj = mso.get_obj('schemas', displayName=schema)
+    if not schema_obj:
+        mso.fail_json(msg="Provided schema '{0}' does not exist".format(schema))
+
+    schema_path = 'schemas/{id}'.format(**schema_obj)
+
+    # Get template
+    templates = [t.get('name') for t in schema_obj.get('templates')]
+    if template not in templates:
+        mso.fail_json(msg="Provided template '{template}' does not exist. Existing templates: {templates}".format(template=template,
+                                                                                                                  templates=', '.join(templates)))
+    template_idx = templates.index(template)
+
+    # Get External EPG
+    externalEpgs = [e.get('name') for e in schema_obj.get('templates')[template_idx]['externalEpgs']]
+    if external_epg not in externalEpgs:
+        mso.fail_json(msg="Provided external epg '{externalEpg}' does not exist. Existing epgs: {externalEpgs}"
+                      .format(externalEpg=external_epg, externalEpgs=', '.join(externalEpgs)))
```

Comment:
Use snake case for externalEpg and externalEpgs so external_epg and external_epgs